web3/pm.py

```diff
@@ -1,27 +1,220 @@
+import json
+
+from eth_utils import (
+    is_canonical_address,
+    to_canonical_address,
+    to_tuple,
+)
+
+from ens import ENS
+from web3.contract import (
+    Contract,
+)
+from web3.exceptions import (
+    PMError,
+)
 from web3.module import (
     Module,
 )
 
 try:
     from ethpm import (
+        ASSETS_DIR,
         Package,
     )
+    from pytest_ethereum.deployer import (
+        Deployer,
+    )
 except ImportError as exc:
     raise ImportError(
         "To use web3's alpha package management features, you must install the "
         "`ethpm` dependency manually: pip install --upgrade ethpm"
     ) from exc
 
 
-# Package Management is currently still in alpha
+# Package Management is currently still in alpha.
 # It is not automatically available on a web3 object.
 # To use the `PM` module, attach it to your web3 object
 # i.e. PM.attach(web3, 'pm')
+
+
 class PM(Module):
     def get_package_from_manifest(self, manifest):
+        """
+        * :manifest: must be a dict representing a valid manifest
+        * Returns a ``Package`` instance representing the Manifest
+        """
         pkg = Package(manifest, self.web3)
         return pkg
 
-    def get_package_from_uri(self, uri):
-        pkg = Package.from_uri(uri, self.web3)
+    def get_package_from_uri(self, manifest_uri):
+        """
+        * :uri: *must* be a valid content-addressed URI, as defined in the
+        `Py-EthPM Documentation <https://py-ethpm.readthedocs.io/en/latest/uri_backends.html>`_.
+        * Returns a ``Package`` isntance representing the Manifest stored at the URI.
+        """
+        pkg = Package.from_uri(manifest_uri, self.web3)
         return pkg
+
+    def set_registry(self, address, ens_address=None):
+        """
+        Sets the current registry used in ``web3.pm`` functions that read/write to an
+        onchain registry.
+
+        :address: Address of the on-chain registry - Accepts ENS name
+        :ens_address: If using an ens, but not on the mainnet, this is the address of the central
+        ENS registry on whatever chain being used.
+        """
+        if is_canonical_address(address):
+            self.registry = Registry(address, self.web3)
+        elif isinstance(address, str):
```

Comment:
Why `isinstance(v, str)`?  Seems like we should require either a bytes address or a checksum'd address here to be consistent with the rest of the library.